core/src/unprocessed_packet_batches.rs

```diff
@@ -368,9 +379,48 @@ pub fn packet_message(packet: &Packet) -> Result<&[u8], DeserializedPacketError>
         .ok_or(DeserializedPacketError::SignatureOverflowed(sig_size))
 }
 
-/// Computes `(addition_fee + base_fee / requested_cu)` for `deserialized_packet`
-fn compute_fee_per_cu(_message: &VersionedMessage, _bank: &Bank) -> u64 {
-    1
+/// Computes `(addition_fee / requested_cu)` if feature tx_wide_compute_cap is enabled,
+/// otherwise return default 0u64
+fn compute_fee_per_cu(message: &VersionedMessage, bank: &Bank) -> f64 {
+    if bank.feature_set.is_active(&tx_wide_compute_cap::id()) {
+        if let Some(sanitized_message) = sanitize_message(message, bank) {
+            let mut compute_budget = ComputeBudget::default();
+            match compute_budget.process_message(&sanitized_message, false, true) {
+                Ok(requested_additional_fee) => {
+                    if let Some(requested_units) = f64::from_u64(compute_budget.max_units) {
+                        let requested_additional_fee = f64::from_u64(requested_additional_fee).unwrap_or(0f64);
```

Comment:
do we have any sanitization to make sure the `requested_additional_fee` isn't too high? This conversion could fail and then we crowd the buffer with a bunch of `requested_additional_fee == 0`

Reply:
commit 5e43496762a9421799626318aaf805a4fb888bf2 simplified `compute_fee_per_cu` that now returns an Option<f64>,